freqtrade/freqtradebot.py

```diff
@@ -1032,7 +1032,7 @@ def execute_sell(self, trade: Trade, limit: float, sell_reason: SellType) -> boo
         trade.sell_reason = sell_reason.value
         # In case of market sell orders the order can be closed immediately
         if order.get('status', 'unknown') == 'closed':
-            trade.update(order)
+            self.update_trade_state(trade, order)
```

Comment:
why should `self.update_trade_state(trade, order)` be used here and `trade.update(order)` is incorrect?

Reply:
Because `trade.update()` should only be used from within `update_trade_state()` - it does all the "fee" updating / handling.

If `trade.update()` is called directly from this location, then the trade is closed without, but the sell-fee is not updated.
It doesn't make a difference if the fee is the "exchange default fee" - but it is a problem if BNB fees are used - or any other rebate somehow applies.

Reply:
`trade.update()` is also called from `handle_stoploss_on_exchange()`, is it okay?